scripts/get-version

```diff
@@ -1,76 +1,80 @@
-#!/bin/sh
+#!/bin/bash
+# shellcheck disable=SC2164
 
-# $1 - os name {linux|darwin|win}
-# $2 - [optional] raw version string "vX.Y-patchN-sha1". as from `git describe'
-#      (see below)
-
-set -eu
-#set -x
+# consistently name all builds, regardless of OS or desktop/mobile
+#
+# we do need to be able to create three digit (M.m.p) and four digit (M.m.p.c) version strings
+# default is VERSION_EXTENSION version - an argument of '4' or '3' gets you a digits only version string
+#
+# we hardcode a base version - this will rarely change
+# (we actually haven't discussed a situation where it would change...)
+SUBSURFACE_BASE_VERSION=6.0
 
+# little silly helper functions
 croak() {
 	echo "$0: $*" >&2
 	exit 1
 }
+croak_usage() {
+	croak "Usage: $0 [3|4]"
+}
+
+if [[ $# -gt 1 ]] ; then croak_usage ; fi
+if [[ $# -eq 1 ]] ; then
+	if [[ $1 != "4" && $1 != "3" ]] ; then croak_usage ; fi
+	DIGITS="$1"
+fi
 
-[ $# -ge 1 ] || croak "missing OS argument"
-os=$1
+# figure out where we are in the file system
+cd "$(dirname "$0")/../"
+export SUBSURFACE_SOURCE=$PWD
 
-if [ $# -eq 2 ] && [ "$2" ]; then
-	v0=$2
+# add the build number to this as 'patch' component
+# if we run in an environment where we are given a build number (e.g. CICD builds)
+# we just grab that - otherwise we have to figure it out on the fly
+if [ ! -f latest-subsurface-buildnumber ] ; then
+	# figure out the most recent build number, given our own git history
+	# this assumes that (a) we are in a checked out git tree and
+	# (b) we have the ability to check out another git repo
+	# in situations where either of these isn't true, it's the caller's
+	# responsibility to ensure that the latest-subsurface-buildnumber file exists
+	if [ ! -d "$SUBSURFACE_SOURCE/nightly-builds" ] ; then
+		git clone https://github.com/subsurface/nightly-builds &> /dev/null || croak "failed to clone nightly-builds repo"
+	fi
+	cd nightly-builds
+	git fetch &> /dev/null
+	LAST_BUILD_BRANCHS=$(git branch -a --sort=-committerdate --list | grep remotes/origin/branch-for | head -50 | cut -d/ -f3)
```

Comment:
\<nitpick>
`LAST_BUILD_BRANCHES`?
\</nitpick>

Reply:
oops, forgot to fix - will force push